rasgotransforms/rasgotransforms/render/environment.py

```diff
@@ -0,0 +1,76 @@
+from jinja2 import Environment
+import re
+from datetime import datetime
+from itertools import combinations, permutations, product
+from typing import Callable, Optional, Dict
+from rasgotransforms.exceptions import RenderException
+
+
+class RasgoEnvironment(Environment):
+    def __init__(self, run_query: Callable, *args, **kwargs):
+        super().__init__(*args, extensions=self.rasgo_extensions, **kwargs)
+        for filter_name, method in self.rasgo_filters.items():
+            self.filters[filter_name] = method
+        self.globals = self.rasgo_globals
+        self.globals['run_query'] = run_query
+
+    @property
+    def rasgo_extensions(self):
+        return ['jinja2.ext.do', 'jinja2.ext.loopcontrols']
+
+    @property
+    def rasgo_globals(self):
+        return {
+            "cleanse_name": cleanse_template_symbol,
+            "raise_exception": raise_exception,
+            "itertools": {"combinations": combinations, "permutations": permutations, "product": product},
+        }
+
+    @property
+    def rasgo_filters(self):
+        return {"todatetime": datetime.fromisoformat}
+
+    def render(
+        self,
+        source_code: str,
+        source_table: str,
+        arguments: dict,
+        source_columns: Optional[Dict[str, Dict[str, str]]] = None,
+        override_globals: Optional[Dict[str, Callable]] = None,
+    ) -> str:
+        """
+        Given the source code of a jinja template, its arguments, and metadata about the source table,
+        this renders the transform sql.
+
+        Source columns can be a mapping of table names to columns (e.g. {table_name: {column_name: column_type}})
```

Comment:
It looks like we're only accepting a dict now (which I agree is the correct answer). May just need to update this comment to reflect that